### ISSUE TITLE
Lock navigation or review mode

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7749]
Steps to reproduce:
1. You have an application where you have to use diferent navigation modes (browse mode, object navigation, and screen review)
2. you need very often the object navigation and the browse mode (for example in Outlook when you are dealing with Calendar including files and long messages)
3. you want to use just the object navigation for more than 3 minutes to do some actions, after that you want to use the screen review to see some quick infos about a certain appointment and if it causes conflicts in your calendar
4. You switch back to object navigation and finally end your work. Then you can fortunately take your small finger from the NVDA key. You go into the browse mode to read your final message and send the appointments and so on.
5. Assume you have the keyboard layout set on desktop instead of laptop

Expected behavior:
NVDA should allow a certain key to lock the certain navigation mode like Browse mode. For example, object navigation could be activated by pressing shift + NVDA + space bar. Repeat this command to return to the original navigation mode. The screen review could be activated by pressing ctrl + shift + NVDA + space bar and can be deactivated by pressing this again. NVDA should allow to change from screen review in object navigation directly. The same for object navigation and browse mode and so on. Only if you press the same command twice then you activate a certain mode and return back to the original one.

Actual behavior:
NVDA actually requires you to hold down several key until you are done with your work in that navigation mode. Apart from browse mode and the usual navigation mode all other modes need key being hold down the whole time. This is quite annoying when you are working a long time in object navigation or so.

System configuration:
NVDA version:
all

NVDA Installed or portable:
both

Other information:

Windows version:
all

Name and version of other software in use when reproducing the issue:
all

Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
yes, all

**Comment by Adriani90** [comment]
#7855

**Comment by Adriani90** [comment]
Slightly edited.

**Comment by Adriani90** [comment]
One further thought which came to my mind.
Instead of implementing a key stroke for each review mode, there could be a review mode ring implemented. (i.e. ideally nvda+space bar could swith between all review modes, including browse mode, focus mode, object navigation, screen review, document review and object review. Once one is active, users would just have to use arrow keys to navigate accordingly. For commands not supported in the specific review mode, NVDA would report "not supported in this review mode".
To limit the burden level on new users, there should be a time limit on pressing nvda+space bar to move through review modes. I.e. pressing nvda+space multiple times with a time lag of let's say 1.5 seconds, NVDA would only switch between browse mode and focus mode. pressing nvda+space bar faster (i.e. less than 1.5 second between space bar presses", NVDA would switch through the review ring.
cc: @leonardder, @lukaszgo1 what do you think?

**Comment by lukaszgo1** [comment]
My understanding of this issue is as follows:
As a user of the laptop layout (the desktop one is less affected as only navigation mode requiring pressing more than one key simultaneously is object navigation) I want to navigate in all modes by pressing as few keys as possible.
@Adriani90 Is my understanding correct? If so I propose to simply make NVDA key blockable similar to what Speakup offers with Speakup key _ z.

**Comment by Adriani90** [comment]
@lukaszgo1 yes your understanding is correct, but locking the NVDA key could cause other issues such as accidentally changing settings you don't want to change. So this lock feature should impact actually only the navigation modes, and the keyboard should behave coresponding to the enabled navigation mode.
